### PR TITLE
Fix overflow issue on renewal

### LINKED ISSUE #1190: certs renew all panics
## Versions

v1.12.0 and master

## What happened?

run `sudo kubeadm alpha phase certs renew all --use-api=false` and then kubeadm panics

/kind bug
/priority critical-urgent
/area ux

### PR DESCRIPTION
**Pull request by liztio**
Co-authored by: Ruben Orduz <rubenoz@gmail.com>

**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:

We were running into some weird issues (namely, stack overflows) when trying to call Cobra from inside Cobra. This fixes the `renew all` logic to not rely on Cobra at all.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#1190

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by timothysc — CHANGES_REQUESTED** [review]
@liztio - this is weird needs more comments and details explaining "why"?

File: cmd/kubeadm/app/cmd/alpha/certs.go [hunk 1]
```diff
@@ -125,27 +124,33 @@ func addFlags(cmd *cobra.Command, cfg *renewConfig) {
 }
 
 // generateCertCommand takes mostly strings instead of structs to avoid using structs in a for loop
-func generateCertCommand(name, longName, baseName, caCertBaseName string, cfg *renewConfig) *cobra.Command {
-	return &cobra.Command{
+func generateCertCommand(name, longName, baseName, caCertBaseName string, cfg *renewConfig) (*cobra.Command, func()) {
```
[the remark this answers is not in the document]
Comment (liztio) [reply cmd/kubeadm/app/cmd/alpha/certs.go]: Do the new changes make more sense?

File: cmd/kubeadm/app/cmd/alpha/certs.go [hunk 2]
```diff
@@ -89,14 +89,20 @@ func getRenewSubCommands() []*cobra.Command {
...
 	for caCert, certs := range certTree {
 		// Don't offer to renew CAs; would cause serious consequences
 		for _, cert := range certs {
-			cmd := makeCommandForRenew(cert, caCert, cfg)
+			// get the cobra.Command skeleton for this command
+			cmd := generateRenewalCommand(cert, cfg)
+			// get the implementation of renewing this certificate
+			renewalFunc := generateRenewalFunction(cert, caCert, cfg)
+			// install the implementation into the command
+			cmd.Run = func(*cobra.Command, []string) { renewalFunc() }
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/alpha/certs.go]: could generateRenewalFunction() return a function with the `func(*cobra.Command, []string)` signature?

**Review by neolit123 — COMMENTED** [review]
thanks @liztio 
added one minor comment.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

This is easier to reason about, thx @liztio

**Pull request merged**